app/src/org/commcare/android/database/migration/FixtureSerializationMigration.java

```diff
@@ -48,16 +55,27 @@ public static boolean migrateFixtureDbBytes(SQLiteDatabase db, Context c) {
         ConcreteAndroidDbHelper helper = new ConcreteAndroidDbHelper(c, db);
         DataInputStream fixtureByteStream = null;
         try {
-            SqlStorage<Persistable> userFixtureStorage =
-                    new SqlStorage<Persistable>("fixture", FormInstance.class, helper);
-            for (SqlStorageIterator i = userFixtureStorage.iterate(false); i.hasMore(); ) {
+            HybridFileBackedSqlStorage<Persistable> fixtureStorage;
+            if (fileMigrationKeySeed != null) {
+                fixtureStorage =
+                        new HybridFileBackedSqlStorageForMigration<Persistable>("fixture", FormInstance.class, helper, baseDir, fileMigrationKeySeed);
+            } else {
+                fixtureStorage =
+                        new UnencryptedHybridFileBackedSqlStorage<Persistable>("fixture", FormInstance.class, helper, baseDir);
+            }
+            SqlStorage<Persistable> oldUserFixtureStorage =
+                    new SqlStorage<Persistable>("oldfixture", FormInstance.class, helper);
+            int count = 0;
+            for (SqlStorageIterator i = oldUserFixtureStorage.iterate(false); i.hasMore(); ) {
                 int id = i.nextID();
+                Log.d(TAG, "migrating fixture " + count++);
```

Comment:
Seems dangerously cute. I'd pull it out, lest someone rely on count in the future, and someone else removes the log statement. Log statements don't feel like they should have side effects.